MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"name", name];
+    [request setPredicate:predicate];
+    
+    NSUInteger count = [self.managedObjectContext countForFetchRequest:request error:&error];
+    if (!error) {
+        return count;
+    } else {
+        return 0;
+    }
+}
+
+- (NSUInteger)checkCountForEntityWithEntityName:(NSString *)entityName withIDNumber:(NSNumber *)entity_id
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %d", @"id", entity_id.intValue];
+    [request setPredicate:predicate];
+    NSUInteger count = [self.managedObjectContext countForFetchRequest:request error:&error];
+    if (!error) {
+        return count;
+    } else {
+        return 0;
+    }
+}
+
+#pragma mark - Boilerplate Core Data Code
+
+@synthesize managedObjectContext = _managedObjectContext;
+@synthesize managedObjectModel = _managedObjectModel;
+@synthesize persistentStoreCoordinator = _persistentStoreCoordinator;
+
+- (NSManagedObjectModel *)managedObjectModel
+{
+    // The managed object model for the application. It is a fatal error for the application not to be able to find and load its model.
+    if (_managedObjectModel != nil) {
+        return _managedObjectModel;
+    }
+    NSURL *modelURL = [[NSBundle mainBundle] URLForResource:@"MH4U_Dex" withExtension:@"momd"];
+    _managedObjectModel = [[NSManagedObjectModel alloc] initWithContentsOfURL:modelURL];
+    return _managedObjectModel;
+}
+
+- (NSPersistentStoreCoordinator *)persistentStoreCoordinator
+{
+    // The persistent store coordinator for the application. This implementation creates and return a coordinator, having added the store for the application to it.
+    if (_persistentStoreCoordinator != nil) {
+        return _persistentStoreCoordinator;
+    }
+    
+    // Create the coordinator and store
+    
+    _persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:[self managedObjectModel]];
+    NSURL *storeURL = [[self applicationDocumentsDirectory] URLByAppendingPathComponent:@"MH4U_Dex.sqlite"];
+    NSError *error = nil;
+    NSString *failureReason = @"There was an error creating or loading the application's saved data.";
+    if (![_persistentStoreCoordinator addPersistentStoreWithType:NSSQLiteStoreType configuration:nil URL:storeURL options:nil error:&error]) {
+        // Report any error we got.
+        NSMutableDictionary *dict = [NSMutableDictionary dictionary];
+        dict[NSLocalizedDescriptionKey] = @"Failed to initialize the application's saved data";
+        dict[NSLocalizedFailureReasonErrorKey] = failureReason;
+        dict[NSUnderlyingErrorKey] = error;
+        error = [NSError errorWithDomain:@"YOUR_ERROR_DOMAIN" code:9999 userInfo:dict];
```

Comment:
You probably should replace `@"YOUR_ERROR_DOMAIN"` with a constant that is exported (so that callers can test the error domain) and with a reverse-DNS-style string (e.g., `io.vokal.something`).